components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/mappings/APIMappingUtil.java

```diff
@@ -2866,6 +2864,23 @@ private static JSONObject handleEndpointSecurity(API api, JSONObject endpointSec
         return endpointSecurityElement;
     }
 
+    /**
+     * Set AWS Secret Key based on preserveCredentials state
+     *
+     * @param awsEndpointConfig Endpoint configuration of the API
+     * @param preserveCredentials Condition to preserve credentials
+     * @return Updated endpoint config
+     */
+    private static JSONObject handleAWSCredentials(JSONObject awsEndpointConfig, boolean preserveCredentials) {
+        if (!preserveCredentials) {
+            if (StringUtils.isNotEmpty((String) awsEndpointConfig.get(APIConstants.AMZN_SECRET_KEY))) {
+                awsEndpointConfig.put(APIConstants.AMZN_SECRET_KEY, APIConstants.AWS_SECRET_KEY);
```

Comment:
if we preserve the AWS credentials in this mode secret key stored in an encrypted format.
this need to be in decrypted format since the CTL project goes between environment where encryption key was different

Reply:
Fixed in 8fe2c6f